src/main/java/uk/gov/hmcts/ccd/domain/service/callbacks/CallbackService.java

```diff
@@ -22,29 +20,71 @@
-
 @Service
 public class CallbackService {
     private static final Logger LOG = LoggerFactory.getLogger(CallbackService.class);
+    public static final int CALLBACK_RETRY_INTERVAL_MULTIPLIER = 3;
 
     private final SecurityUtils securityUtils;
-    private final RestTemplate restTemplate;
+    private final List<Integer> defaultCallbackRetryIntervalsInSeconds;
+    private final Integer defaultCallbackTimeoutInMillis;
+    private final RestTemplateProvider restTemplateProvider;
+
+    static class CallbackRetryContext {
+        private final Integer callbackRetryInterval;
+        private final Integer callbackRetryTimeout;
+
+        CallbackRetryContext(final Integer callbackRetryInterval, final Integer callbackRetryTimeout) {
+            this.callbackRetryInterval = callbackRetryInterval;
+            this.callbackRetryTimeout = callbackRetryTimeout;
+        }
+
+        Integer getCallbackRetryInterval() {
+            return callbackRetryInterval;
+        }
+
+        Integer getCallbackRetryTimeout() {
+            return callbackRetryTimeout;
+        }
+
+    }
 
     @Autowired
     public CallbackService(final SecurityUtils securityUtils,
-                           @Qualifier("restTemplate") final RestTemplate restTemplate) {
+                           final ApplicationParams applicationParams,
+                           final RestTemplateProvider restTemplateProvider) {
         this.securityUtils = securityUtils;
-        this.restTemplate = restTemplate;
+        this.defaultCallbackRetryIntervalsInSeconds = applicationParams.getCallbackRetryIntervalsInSeconds();
+        this.defaultCallbackTimeoutInMillis = applicationParams.getCallbackReadTimeoutInMillis();
+        this.restTemplateProvider = restTemplateProvider;
+    }
+
+    public Optional<CallbackResponse> send(final String url,
+                                           final List<Integer> callbackRetryTimeouts,
+                                           final CaseEvent caseEvent,
+                                           final CaseDetails caseDetails) {
+        return send(url, callbackRetryTimeouts, caseEvent, null, caseDetails);
     }
 
-    // The retry will be on seconds T=1 and T=3 if the initial call fails at T=0
-    @Retryable(value = {CallbackException.class}, maxAttempts = 3, backoff = @Backoff(delay = 1000, multiplier = 3))
     public Optional<CallbackResponse> send(final String url,
+                                           final List<Integer> callbackRetryTimeouts,
+                                           final CaseEvent caseEvent,
+                                           final CaseDetails caseDetailsBefore,
+                                           final CaseDetails caseDetails) {
+
+        return send(url, callbackRetryTimeouts, caseEvent, caseDetailsBefore, caseDetails, false);
+    }
+
+    @SuppressWarnings("javasecurity:S5145")
+    public Optional<CallbackResponse> send(final String url,
```

Comment:
Can't we use the following send method using generics and call getBody() on the result?